lib/internal/stream_base_commons.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const { Buffer } = require('buffer');
+const errors = require('internal/errors');
+const { WriteWrap } = process.binding('stream_wrap');
+
+const errnoException = errors.errnoException;
+
+function handleWriteReq(req, data, encoding) {
+  const { handle } = req;
+
+  switch (encoding) {
+    case 'buffer':
+      return handle.writeBuffer(req, data);
+    case 'latin1':
+    case 'binary':
+      return handle.writeLatin1String(req, data);
+    case 'utf8':
+    case 'utf-8':
+      return handle.writeUtf8String(req, data);
+    case 'ascii':
+      return handle.writeAsciiString(req, data);
+    case 'ucs2':
+    case 'ucs-2':
+    case 'utf16le':
+    case 'utf-16le':
+      return handle.writeUcs2String(req, data);
+    default:
+      return handle.writeBuffer(req, Buffer.from(data, encoding));
+  }
+}
+
+function createWriteWrap(handle, oncomplete) {
+  const req = new WriteWrap();
+
+  req.handle = handle;
+  req.oncomplete = oncomplete;
+  req.async = false;
+
+  return req;
+}
+
+function writevGeneric(self, req, data, cb) {
+  const allBuffers = data.allBuffers;
+  let chunks;
+  let i;
```

Comment:
Can you restore these as `var`? I'm not sure since when `let` variables are on par with `var` variables for tight loop performance, but this is one of the hottest path in core, I would prefer if we used very fast ES5

Reply:
cc @bmeurer

Reply:
@mcollina I might be wrong, but I think it shouldn’t make a difference for variables that are at the top level of a function?

Reply:
const or let have some overhead in Node 8 and 9 in tight loops such as this one. I’m not sure if this still the case with current master.

I’d prefer if this refactoring was separated from changing from var to const/let.

Reply:
@mcollina makes sense, i'll change them back to `var`